app/models/agents/google_translation_agent.rb

```diff
@@ -40,17 +48,19 @@ def translate_from
     end
 
     def receive(incoming_events)
-      translate = Google::Apis::TranslateV2::TranslateService.new
-      translate.key = ENV['GOOGLE_API_KEY']
       incoming_events.each do |event|
         translated_event = {}
         opts = interpolated(event)
         opts['content'].each_pair do |key, value|
-          result = translate.list_translations(value, opts['to'], source: translate_from)
+          result = translate_service.list_translations(value, opts['to'], source: translate_from)
           translated_event[key] = result.translations.last.translated_text
         end
-        create_event :payload => translated_event
+        create_event payload: translated_event
       end
     end
+
+    def translate_service
+      @translate_service ||= Google::Apis::TranslateV2::TranslateService.new.tap {|service| service.key = options['google_api_key']}
+    end
```

Comment:
I noticed that you memoize the method `translate_service` I agree it's better for performance but in our case the method has `options['google_api_key']` as parameter. What do you think about using a hash with the value of `options['google_api_key']` as keys in case a user changes the API key.